pkg/transforms/policy.go

```diff
@@ -130,16 +130,172 @@ func OperatorPolicyResourceBuilder(c *unstructured.Unstructured) *PolicyResource
+	})
+
+	node.Metadata["relObjs"] = objList
+
+	return node
+}
+
+func parseRelatedObject(item any) *policyRelObj {
+	relObj, ok := item.(map[string]any)
+	if !ok {
+		return nil
+	}
+
+	object, found, err := unstructured.NestedMap(relObj, "object")
+	if !found || err != nil {
+		return nil
+	}
+
+	meta, found, err := unstructured.NestedStringMap(object, "metadata")
+	if !found || err != nil {
+		return nil
+	}
+
+	kind, ok := object["kind"].(string)
+	if !ok {
+		return nil
+	}
+
+	return &policyRelObj{
+		kind:      kind,
+		namespace: meta["namespace"],
+		name:      meta["name"],
+	}
+}
+
+func CertPolicyResourceBuilder(c *unstructured.Unstructured) *PolicyResource {
+	node := transformCommon(c) // Start off with the common properties
+
+	detailMap, found, err := unstructured.NestedMap(c.Object, "status", "compliancyDetails")
+	if len(detailMap) == 0 || !found || err != nil {
+		return &PolicyResource{
+			node: getPolicyCommonProperties(c, node),
+		}
+	}
+
+	certList := []policyRelObj{}
+	for namespace, item := range detailMap {
+		details, ok := item.(map[string]any)
+		if !ok {
+			continue
+		}
+
+		// this "list" is actually a map
+		nonCompCerts, found, err := unstructured.NestedMap(details, "nonCompliantCertificatesList")
+		if len(nonCompCerts) == 0 || !found || err != nil {
+			continue
+		}
+
+		for _, item := range nonCompCerts {
+			cert, ok := item.(map[string]any)
+			if !ok {
+				continue
+			}
+
+			name, ok := cert["secretName"].(string)
+			if !ok {
+				continue
+			}
+
+			certList = append(certList, policyRelObj{
+				kind:      "Secret",
+				namespace: namespace,
+				name:      name,
+			})
+		}
+	}
+
+	slices.SortFunc(certList, func(a, b policyRelObj) int {
+		return strings.Compare(a.kind+a.namespace+a.name, b.kind+b.namespace+b.name)
+	})
+
+	node.Metadata["relObjs"] = certList
+
+	return &PolicyResource{
+		node: getPolicyCommonProperties(c, node),
+	}
 }
 
 // BuildNode construct the node for Policy Resources
 func (p PolicyResource) BuildNode() Node {
 	return p.node
 }
 
-// BuildEdges construct the edges for Policy Resources
 func (p PolicyResource) BuildEdges(ns NodeStore) []Edge {
-	// no op for now to implement interface
-	return []Edge{}
+	kindRaw, ok := p.node.Properties["kind"]
+	if !ok {
+		return []Edge{}
+	}
+
+	policyKind, ok := kindRaw.(string)
+	if !ok {
+		return []Edge{}
+	}
+
+	if !p.node.hasMetadata("relObjs") {
+		return []Edge{}
+	}
+
+	relatedObjects, ok := p.node.Metadata["relObjs"].([]policyRelObj)
+	if !ok {
+		return []Edge{}
+	}
+
+	edges := make([]Edge, 0, len(relatedObjects))
+
+	for _, obj := range relatedObjects {
+		namespace := obj.namespace
+
+		if obj.namespace == "" {
+			namespace = "_NONE"
+		}
+
+		if res, ok := ns.ByKindNamespaceName[obj.kind][namespace][obj.name]; ok {
+			edges = append(edges, Edge{
+				EdgeType:   "relatedResource",
+				SourceKind: policyKind,
+				SourceUID:  p.node.UID,
+				DestKind:   obj.kind,
+				DestUID:    res.UID,
+			})
+		}
+	}
```

Comment:
Please move this to a generic function, so we can reuse it from other transforms. See this comment. https://github.com/stolostron/search-collector/pull/595/files#r2068962456